### PR TITLE
Add seccomp validation tests

### PR DESCRIPTION
**Pull request by feiskyer**
Part of #24: Add seccomp validation tests.

### CODE REVIEW

**Comment by feiskyer** [comment]
CI failed because dockerd installed doesn't support seccomp:

seccomp profiles are not supported on this daemon, you cannot specify a custom seccomp profile.

File: hack/install-docker.sh [hunk 1]
```diff
@@ -18,10 +18,20 @@ set -o errexit
...
+# Install docker.
 sudo sh -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-$(lsb_release -cs) main" > /etc/apt/sources.list.d/docker.list'
 curl -fsSL https://apt.dockerproject.org/gpg | sudo apt-key add -
 sudo apt-key fingerprint 58118E89F3A912897C070ADBF76221572C52609D
 sudo apt-get update
-sudo apt-get -y install "docker-engine=1.13.1-0~ubuntu-$(lsb_release -cs)"
+sudo apt-get -y install "docker-engine=17.03.1~ce-0~ubuntu-$(lsb_release -cs)"
+
+# docker debs don't support seccomp, so we install a static version instead.
+curl -sSL -o docker.tgz https://download.docker.com/linux/static/stable/x86_64/docker-17.03.1-ce.tgz
```
Comment (Random-Liu) [inline hack/install-docker.sh]: Cool! :) Ref https://github.com/moby/moby/issues/22870

File: hack/install-docker.sh [hunk 2]
```diff
@@ -18,10 +18,20 @@ set -o errexit
...
 
-# Install docker
+
+# Install docker.
 sudo sh -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-$(lsb_release -cs) main" > /etc/apt/sources.list.d/docker.list'
 curl -fsSL https://apt.dockerproject.org/gpg | sudo apt-key add -
 sudo apt-key fingerprint 58118E89F3A912897C070ADBF76221572C52609D
 sudo apt-get update
-sudo apt-get -y install "docker-engine=1.13.1-0~ubuntu-$(lsb_release -cs)"
+sudo apt-get -y install "docker-engine=17.03.1~ce-0~ubuntu-$(lsb_release -cs)"
```
Comment (Random-Liu) [inline hack/install-docker.sh]: Do we still need this given that we are using static version now.

**Comment by Random-Liu** [comment]
@feiskyer LGTM with one question.

File: hack/install-docker.sh (hunk 2 above)
Comment (feiskyer) [reply hack/install-docker.sh]: this is for manage docker service, e.g. we could restart docker with `service docker restart` command. Or else, we need to manage service by ourself, e.g. if docker is started in first stage, we need to kill dockerd and start again in second stage.

Comment (Random-Liu) [reply hack/install-docker.sh]: Another question is that why do we use docker 17.03? There is no available 1.13 static binary?

Comment (feiskyer) [reply hack/install-docker.sh]: yep, didn't found 1.13 static binary. 17.03 is the oldest one and also will be supported in kuberentes 1.8.

Comment (Random-Liu) [reply hack/install-docker.sh]: @feiskyer I don't think `17.03` is supported in Kubernetes 1.8, but yeah I also checked, 17.03 is the oldest one.

I'm fine with testing it for now. We'll be setting up CRI validation in Kubernetes test infra soon, we could test against docker 1.13 then.

Comment (feiskyer) [reply hack/install-docker.sh]: >  I don't think 17.03 is supported in Kubernetes 1.8

@Random-Liu See https://github.com/kubernetes/kubernetes/issues/42926.

Comment (Random-Liu) [reply hack/install-docker.sh]: If so, it means that https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/system/types.go#L162 needs to be updated.

@yguo0905 told me that docker 1.13 is the newest version.

Comment (yguo0905) [reply hack/install-docker.sh]: Sorry, I should have mentioned that they are the same minor release. Created https://github.com/kubernetes/kubernetes/pull/52666 to add 17.03 support to the spec.

**Pull request merged**